src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,266 @@
+
+  @Argument(fullName = StandardArgumentDefinitions.GENOMICSDB_ARRAY_LONG_NAME,
+  shortName = StandardArgumentDefinitions.GENOMICSDB_ARRAY_SHORT_NAME,
+  doc = "TileDB array name used by GenomicsDB",
+  optional = true)
+  private String arrayName = DEFAULT_ARRAY_NAME_PREFIX;
+
+  @Argument(fullName = StandardArgumentDefinitions.GENOMICSDB_SEGMENT_SIZE_LONG_NAME,
+    shortName = StandardArgumentDefinitions.GENOMICSDB_SEGMENT_SIZE_SHORT_NAME,
+    doc = "Buffer size in bytes allocated for GenomicsDB attributes during " +
+      "import. Should be large enough to hold one variant data for one sample",
+    optional = true)
+  private
+  long segmentSize = DEFAULT_SEGMENT_SIZE;
+
+  @Argument(fullName = StandardArgumentDefinitions.VARIANT_LONG_NAME,
+  shortName = StandardArgumentDefinitions.VARIANT_SHORT_NAME,
+  doc = "GVCF files to be imported to GenomicsDB")
+  private List<String> variantFileNames;
+
+  @Argument(fullName = StandardArgumentDefinitions.SIZE_PER_COLUMN_PARTITION_PER_SAMPLE_LONG_NAME,
+  shortName = StandardArgumentDefinitions.SIZE_PER_COLUMN_PARTITION_PER_SAMPLE_SHORT_NAME,
+  doc = "Buffer size in bytes allocated to GenomicsDB variant readers. Default is 16KBytes." +
+    " Larger values are better as smaller values cause frequent disk writes",
+  optional = true)
+  private long sizePerColumnPartitionPerSample = DEFAULT_SIZE_PER_COLUMN_PARTITION_PER_SAMPLE;
+
+  @Argument(fullName = StandardArgumentDefinitions.GENOMICSDB_VID_MAP_FILE_LONG_NAME,
+  shortName = StandardArgumentDefinitions.GENOMICSDB_VID_MAP_FILE_SHORT_NAME,
+  doc = "JSON file including INFO/FORMAT/FILTER fields from headers and contig maps")
+  private String vidMapJSONFile = "";
+
+  @Argument(fullName = StandardArgumentDefinitions.GENOMICSDB_CALLSET_MAP_FILE_LONG_NAME,
+    shortName = StandardArgumentDefinitions.GENOMICSDB_CALLSET_MAP_FILE_SHORT_NAME,
+    doc = "JSON file including callset to GenomicsDB row maps")
+  private String callsetMapJSONFile = "";
+
+  @Argument(fullName = StandardArgumentDefinitions.GENOMICSDB_CREATE_WORKSPACE_LONG_NAME,
+  shortName = StandardArgumentDefinitions.GENOMICSDB_CREATE_WORKSPACE_SHORT_NAME,
+  doc = "Boolean flag - true means it will create GenomicsDB workspace, otherwise not",
+  optional = true)
+  private Boolean createWorkspace = false;
+
+  @Override
+  public boolean requiresIntervals() { return true; }
+
+  @Override
+  public boolean requiresReads() { return false; }
+
+  @Override
+  public boolean requiresReference() {
+    return false;
+  }
+
+  // The GenomicsDB importer object
+  private GenomicsDBImporter importer = null;
+
+  // Intervals from command line (singleton for now)
+  private List<ChromosomeInterval> intervals;
+
+  /**
+   * From the command line validate
+   * 1. GenomicsDB workspace exists
+   * 2. GenomicsDB workspace is a valid one
+   *
+   * @return  Error messages for the exceptions
+   */
+  @Override
+  protected String[] customCommandLineValidation() {
```

Comment:
Move remaining checks from `customCommandLineValidation()` to `onTraversalStart()`, and get rid of the override of `customCommandLineValidation()`